src/components/ActivityForm/FormInput.tsx

```diff
@@ -409,7 +529,12 @@ const FormInput: React.FC = () => {
       </div>
 
       <div className="flex justify-between items-center cursor-pointer my-9">
-        <button onClick={() => dispatch(setStep('selection'))}>Back</button>
+        {isEditing ? (
```

Comment:
we can do this with a single button, and put the `isEditing?` logic inside the `onClick`:

```
<button onClick={() => isEditing ? router.back() : dispatch(setStep('selection'))}>Back</button>
```